### DISCUSSION TITLE
Should M³ provide beta versions?

### DISCUSSION THREAD

**Discussion by mtdvlpr** [sircharlo/meeting-media-manager#1526] [Polls]
I have been wondering. Some appreciate it when their feature request or bug fix is available before the official release comes out. Currently, these beta/test versions are manually provided by downloading the artifacts from GitHub and uploading it to Google Drive. We could invest some time in trying to implement a (semi) automatic way of providing everyone with the latest changes in the form of a beta release. Before we do, I'd like to know if that's something that you guys want.

**Comment by Hi-there-how-are-u** [comment]
Yes, it would be great when translating to see if I got it right...

**Comment by e-Feitosa** [comment]
A suggestion: The beta version could only be available to those who indicate the beta option in the M3. This would make testing easier. For example, I always test it at home, and then install it on the KH. If you can choose to use or not use the beta version, that would be great.

**Reply by mtdvlpr** [reply]
Yeah, I'm envisioning a toggle in the settings that would allow you to subscribe to the beta releases for auto updates. Still have to fogure out how to though.

**Reply by sircharlo** [reply]
Seems [simple enough](https://www.electron.build/tutorials/release-using-channels.html).

**Reply by mtdvlpr** [reply]
Yeah, I found that too, but ideally I'd like to automate it to release a new beta release for every commit to the master branch.

**Reply by sircharlo** [reply]
Of course you do 😆

**Reply by mtdvlpr** [reply]
I figured it out! See my fork. It bumps the package.json version (23.2.1 -> 23.2.2-beta.0 -> 23.2.2-beta.1), creates a draft, builds the app, publishes the draft. It executes on every commit, but fails on the `chore(release)` commit. We can also let it run every x hours/days.

**Comment by Chipperchap** [comment]
As regards the vote - On my home PC I don't mind beta versions as I trial and test and become familiar with the latest features using my home setup. At the Kingdom Hall of course we would only use stable releases. 😁

**Reply by mtdvlpr** [reply]
Completely valid! I currently have a nice automation in the works that would release a new beta version every half hour (if there are new changes to release) which you would automatically receive by turning a `receive beta releases` option on. I'm pretty positive we'll implement this feature.

**Reply by Hi-there-how-are-u** [reply]
Can this beta be turned off by the VO when using the syncrothingie?
That would be great so that it would not be visible at all inte Kingdom hall.

**Reply by sircharlo** [reply]
> Can this beta be turned off by the VO when using the syncrothingie? That would be great so that it would not be visible at all inte Kingdom hall.

I'm guessing not, since this is a per computer setting.

**Reply by mtdvlpr** [reply]
We could make it lockable, but then no one connected to the congregation server can enable it for themselves. Not sure that's what you want.

**Reply by Hi-there-how-are-u** [reply]
Most likely it will not be a problem. Just need to inform all users at KH that this is a no go... Try it home but not at KH.
Lot of abbreviations here... VO, KH and the Congregation sync server should be CSS, not confusing at all! :)

**Comment by mtdvlpr** [comment]
Good news! We've implemented the beta release feature! A beta version with the latest changes will automatically be released every 12 hours. You can opt in to these releases using a toggle in the App Configuration settings. When enabling this option, you might need to restart the application to start receiving the beta updates.

**Reply by e-Feitosa** [reply]
> Boas notícias! Implementamos o recurso de versão beta! Uma versão beta com as últimas alterações será lançada automaticamente a cada 12 horas. Você pode optar por essas versões usando uma alternância nas definições de Configuração do Aplicativo. Ao habilitar essa opção, talvez seja necessário reiniciar o aplicativo para começar a receber as atualizações beta.


@mtdvlpr, About the beta version, is it possible to know [on the M3 screen] what is being tested? Or just following here on the forum? When you click to check the information after the update, it goes to the current default version releases screen.

**Reply by mtdvlpr** [reply]
@e-Feitosa, [here](https://github.com/sircharlo/meeting-media-manager/releases/tag/v23.2.3-beta.0) is the latest beta information (nothing exciting).

The 'more information' button link will be fixed in the next release :)

**Comment by e-Feitosa** [comment]
@mtdvlpr, in the beta version, I didn't find the option to test the zoom meeting integration. And when I installed the version that is Google Drive, the M3 downloads an update and removes the zoom+m3 test option. Would it be possible to include it for testing too, or is it still too "early"?

**Reply by mtdvlpr** [reply]
@e-Feitosa, the automatic beta includes features that have been merged already. Since the Zoom integration isn't merged yet, it's not included. I'll update the Google Drive test version, so that it doesn't want to update.

**Reply by mtdvlpr** [reply]
The Zoom integration test version has been updated :)
